### PR TITLE
feat(chart): year seasons toggle [#15]

### PR DESCRIPTION
**Pull request by davidholyko**
#### Summary

- Work Type: Feature 
- Fixes: https://github.com/starter-code/binge/issues/15

#### What I did

- 
- 
- 

#### Notes / Research / References

- Need to fix episodeLabels for year/episode
- 2

#### Before/After Preview

- Before:
- After: https://deploy-preview-29--binge.netlify.app/#/search
  - Search Arrested Development, Click Toggle
  - Screen Shot 2020-07-06 at 10 38 59 PM

#### Checklist

- [x] Branch is in format `TYPE/scope/description`
- [x] PR title is in format `TYPE(scope): description`
- [x] Commits are in semantic format
- [x] Has Summary

### CODE REVIEW

File: src/components/Chart/ChartSquare.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ export const ChartSquare = ({ data, type }) => {
   };
 
   const style = {
-    gridColumnStart: episodeNumber,
+    gridColumnStart: type === 'yearLabels' ? episodeNumber : null,
```
Comment (davidholyko) [inline src/components/Chart/ChartSquare.js:31]: fixes bug where year number is not directly to the left of first episode in that season

File: src/components/Chart/ChartSquare.js [hunk 2]
```diff
@@ -40,6 +40,7 @@ export const ChartSquare = ({ data, type }) => {
     'episode-data-square': type === 'episodeData',
```
Comment (davidholyko) [inline src/components/Chart/ChartSquare.js:40]: nit make Object instead of string literals

File: src/components/Chart/Chart.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import { _ } from '../../utils';
-import React from 'react';
+import React, { useState } from 'react';
 
 import { ChartSquare } from './ChartSquare';
 
 export const Chart = ({ data: chartInfo }) => {
+  const [chartType, setChartType] = useState('season/episode');
```
Comment (davidholyko) [inline src/components/Chart/Chart.js:7]: nit use object instead of string literals

**Comment by davidholyko** [comment]
We might be at the point where we should move the label generation to back end. the original idea of the front end was that it only did displaying data and no calculations.

File: src/components/Chart/Chart.js [hunk 4]
```diff
@@ -63,14 +99,34 @@ export const Chart = ({ data: chartInfo }) => {
     return output;
   });
 
-  const newChartInfo = [...chartData, ...seasonLabels, ...episodeLabels];
+  const nChartData = _.map(nData, (data) => {
```
Comment (simonchen92) [inline src/components/Chart/Chart.js:102]: we may want to rename this in the future to be more semantic for users looking at our code

**Pull request merged**